internal/dcache/rpc/server/handler.go

```diff
@@ -47,43 +57,894 @@ var _ service.ChunkService = &ChunkServiceHandler{}
+// call cluster manager helper method to get available disk space for the given path
+func getAvailableDiskSpace(path string) (int64, error) {
+	return 0, nil
+}
+
+// TODO:: integration: sample method, will be later removed after integrating with cluster manager
+// call cluster manager helper method to get the node ID of this node
+func getNodeUUID() string {
+	return "node-uuid" // TODO: get the node uuid from the config
+}
+
+// check the if the chunk address is valid
+// - check if the rvID is valid
+// - check if the cache dir exists
+// - check if the MV is valid
+func (h *ChunkServiceHandler) checkValidChunkAddress(address *models.Address) error {
+	if address == nil || address.FileID == "" || address.RvID == "" || address.MvName == "" {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: Invalid chunk address %+v", *address)
+		return rpc.NewResponseError(rpc.InvalidRequest, fmt.Sprintf("invalid chunk address %+v", *address))
+	}
+
+	// check if the rvID is valid
+	rvInfo, ok := h.rvIDMap[address.RvID]
+	if !ok || rvInfo == nil {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: Invalid rvID %s", address.RvID)
+		return rpc.NewResponseError(rpc.InvalidRVID, fmt.Sprintf("invalid rvID %s", address.RvID))
+	}
+
+	cacheDir := rvInfo.cacheDir
+	if cacheDir == "" || !common.DirectoryExists(cacheDir) {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: Cache dir not found for RV %s", rvInfo.rvName)
+		return rpc.NewResponseError(rpc.InternalServerError, fmt.Sprintf("cache dir not found for RV %s", rvInfo.rvName))
+	}
+
+	// check if the MV is valid
+	mvPath := filepath.Join(cacheDir, address.MvName)
+	if err := rvInfo.isMvValid(mvPath); err != nil {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: MV %s is not hosted by RV %s [%s]", address.MvName, rvInfo.rvName, err.Error())
+		return rpc.NewResponseError(rpc.MVNotHostedByRV, fmt.Sprintf("MV %s is not hosted by RV %s [%s]", address.MvName, rvInfo.rvName, err.Error()))
+	}
+
+	return nil
+}
+
+// get the RVInfo from the RV name
+func (h *ChunkServiceHandler) getRVInfoFromRvName(rvName string) *rvInfo {
+	var rvInfo *rvInfo
+	for _, info := range h.rvIDMap {
+		if info == nil {
+			continue
+		}
+		if info.rvName == rvName {
+			rvInfo = info
+			break
+		}
+	}
+
+	return rvInfo
+}
+
+func (h *ChunkServiceHandler) createMVDirectory(path string) error {
+	log.Debug("ChunkServiceHandler::createMVDirectory: Creating MV directory %s", path)
+	flock := h.locks.Get(path) // TODO: check if lock is needed in directory creation
+	flock.Lock()
+	defer flock.Unlock()
+
+	// TODO: dir check is not needed as os.MkdirAll does nothing if the dir already exists
+	if !common.DirectoryExists(path) {
+		err := os.MkdirAll(path, 0700)
+		if err != nil {
+			log.Err("ChunkServiceHandler::createMVDirectory: Failed to create MV directory %s [%v]", path, err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
don't need lock or the exist check, simply do

```
func (h *ChunkServiceHandler) createMVDirectory(path string) error {
    log.Debug("ChunkServiceHandler::createMVDirectory: Creating MV directory %s", path)

    if err := os.MkdirAll(path, 0700); err != nil && !os.IsExist(err) {
        return fmt.Errorf("MkdirAll(%s) failed: %v", path, err)
    }

    return nil
}
```

Reply:
Done, updated